### DISCUSSION TITLE
Dependency Injection in Avalonia

### DISCUSSION THREAD

**Discussion by cerobe** [AvaloniaUI/Avalonia#14632] [Q&A] [answered]
How is handle Dependency Injection in Avalonia? Is there a recommended approach?

The documentation is "in preparation": https://docs.avaloniaui.net/docs/guides/implementation-guides/how-to-implement-dependency-injection

Can you provide more info or an example?

I'm happy to write the documentation myself and make a PR, I just need some guidance on what the Avalonia community is doing or aligned on.

**Comment by thevortexcloud** [comment, marked as answer]
There is already a PR to add docs, but as I mentioned in that PR it has many issues.

https://github.com/AvaloniaUI/avalonia-docs/pull/339

But seeing as they don't seem to have actually put in any updates since, you may be able to do a separate PR/fix the existing one and put in proper docs (hopefully not written by AI that can't write consistent clean code).

> How is handle Dependency Injection in Avalonia? Is there a recommended approach?

It depends on what you are doing. For simple apps, you just create your container during the framework initialisation and then resolve your main view model from it and set it as your startup view's datacontext. For more complex apps, you might need an approach like what was discussed here: https://github.com/AvaloniaUI/Avalonia/discussions/13455#discussioncomment-7464329 (make sure you read everything after that comment as there was some follow up discussions).

**Reply by cerobe** [reply]
Thanks, these links were enough, I'm going to make a PR

**Reply by cerobe** [reply]
I've done the PR https://github.com/AvaloniaUI/avalonia-docs/pull/359 if you could have a look, your feedback will be appreciated.
